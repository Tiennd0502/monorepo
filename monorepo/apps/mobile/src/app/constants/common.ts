export const CONGRATS_IMAGE = 'https://i.ibb.co/nkf6hFr/congrats.png';
