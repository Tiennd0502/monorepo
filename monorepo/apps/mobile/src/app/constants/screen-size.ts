import { Dimensions } from 'react-native';

export const BASE_SCREEN_WIDTH = 412;
export const BASE_SCREEN_HEIGHT = 842;

export const SCREEN_WIDTH: number = Dimensions.get('window').width;
export const SCREEN_HEIGHT: number = Dimensions.get('window').height;

export const CONTENT_WIDTH: number = SCREEN_WIDTH - 40;
