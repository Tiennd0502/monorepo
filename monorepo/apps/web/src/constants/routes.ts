export const ROUTES = {
  HOME: '/',
  VERIFY_OTP: '/verify-otp',
  SIGN_UP: '/sign-up',
  LOGIN: '/login',
};
