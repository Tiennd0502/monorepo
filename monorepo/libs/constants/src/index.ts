export * from './apis';
export * from './message';
export * from './commons';
export * from './product';
export * from './category';
export * from './validation';
