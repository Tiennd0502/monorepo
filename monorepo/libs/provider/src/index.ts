export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
