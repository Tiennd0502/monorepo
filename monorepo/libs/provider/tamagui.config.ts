import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
