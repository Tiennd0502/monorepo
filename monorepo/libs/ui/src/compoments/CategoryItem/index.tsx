import { memo } from 'react';

// Types
import { Category } from '@monorepo/types';

// Components
import { Image, Stack } from 'tamagui';
import Text from '../Text';
import IconButton from '../IconButton';

interface CategoryItemProps {
  isActive?: boolean;
  item: Category;
  onPress?: () => void;
}

const CategoryItem = ({
  item: { label, image },
  isActive = false,
  onPress,
}: CategoryItemProps) => (
  <Stack alignContent="center" alignItems="center" gap="$2" onPress={onPress}>
    <IconButton
      onPress={onPress}
      padding="$2"
      borderRadius="$3"
      backgroundColor={isActive ? '$primary' : '$backgroundDisabled'}
    >
      <Image
        source={{ uri: image }}
        width={20}
        height={20}
        alt={label}
        tintColor="$secondary"
      />
    </IconButton>
    <Text color={isActive ? '$primary' : '$textDefault'}>{label}</Text>
  </Stack>
);

export default memo(CategoryItem);
