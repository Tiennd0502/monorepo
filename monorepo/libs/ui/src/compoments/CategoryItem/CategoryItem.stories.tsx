import { Meta, StoryObj } from '@storybook/react';

import { CATEGORIES } from '@monorepo/constants';
import CategoryItem from '.';

const meta: Meta<typeof CategoryItem> = {
  title: 'Components/CategoryItem',
  component: CategoryItem,
};

export default meta;

type Story = StoryObj<typeof CategoryItem>;

export const Default: Story = {
  args: {
    item: CATEGORIES[0],
  },
};

export const Active: Story = {
  args: {
    item: CATEGORIES[0],
    isActive: true,
  },
};
