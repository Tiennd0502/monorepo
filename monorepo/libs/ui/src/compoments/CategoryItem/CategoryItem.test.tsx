import {fireEvent, render, waitFor} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import {CATEGORIES} from '@monorepo/constants';
import CategoryItem from '.';

const onPress = jest.fn();

const baseProps = {
  item: CATEGORIES[0],
  onPress,
};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <CategoryItem {...props} />
    </TamaguiProvider>
  );
};

describe('CategoryItem', () => {
  const onPress = jest.fn();
  const props = {
    item: CATEGORIES[0],
    onPress,
  };

  it('should render CategoryItem component', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render CategoryItem component with props isActive', () => {
    const container = setup({isActive: true});

    expect(container).toMatchSnapshot();
  });

  it('calls onPress when category clicked', () => {
    const {getByTestId} = setup();

    waitFor(() => {
      const item = getByTestId('category');
      fireEvent.press(item);
  
      expect(onPress).toHaveBeenCalled();
    })
  });
});
