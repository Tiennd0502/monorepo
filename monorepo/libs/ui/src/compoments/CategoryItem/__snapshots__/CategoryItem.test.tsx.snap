// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryItem should render CategoryItem component 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "gap": 8,
    }
  }
  testID="category"
>
  <View
    accessibilityRole="button"
    focusVisibleStyle={{}}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#909090",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 0,
        "borderRightColor": "transparent",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": "fit-content",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <Image
      alt="Polular"
      source={
        {
          "uri": "https://i.ibb.co/PQztKcN/star-1.png",
        }
      }
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#999",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 14,
        "lineHeight": 20,
      }
    }
    suppressHighlighting={true}
  >
    Polular
  </Text>
</View>
`;

exports[`CategoryItem should render CategoryItem component with props isActive 1`] = `
<View
  style={
    {
      "alignContent": "center",
      "alignItems": "center",
      "gap": 8,
    }
  }
  testID="category"
>
  <View
    accessibilityRole="button"
    focusVisibleStyle={{}}
    focusable={true}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#242424",
        "borderBottomColor": "transparent",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderBottomWidth": 0,
        "borderLeftColor": "transparent",
        "borderLeftWidth": 0,
        "borderRightColor": "transparent",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "transparent",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "borderTopWidth": 0,
        "flexDirection": "row",
        "flexWrap": "nowrap",
        "height": "fit-content",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <Image
      alt="Polular"
      source={
        {
          "uri": "https://i.ibb.co/PQztKcN/star-1.png",
        }
      }
      style={
        {
          "height": 20,
          "width": 20,
        }
      }
    />
  </View>
  <Text
    allowFontScaling={false}
    style={
      {
        "color": "#242424",
        "fontFamily": "NunitoSans_7pt-Medium",
        "fontSize": 14,
        "lineHeight": 20,
      }
    }
    suppressHighlighting={true}
  >
    Polular
  </Text>
</View>
`;
